test/api_tests.lua

```diff
@@ -467,5 +467,42 @@ test:is(res[1], true, "Schema created successfully")
 res = schema.export(res[2])
 test:is_deeply(res, deferred_canonical, "Exported schema should be canonical.")
 
+local nullable_orig = [[ {
+"name": "outer", "type": "record", "fields":
+    [{ "name": "r1", "type":
+        {"name": "tr1", "type": "record", "fields":
+            [{"name": "v1", "type": "int"} ,
+            {"name": "v2", "type": "string*"} ] } },
+    { "name": "r2", "type": "tr1*"},
+    { "name": "dummy", "type": {
+         "name": "td", "type": "array", "items": "int" }},
+    { "name": "r3", "type": {
+          "name": "tr2", "type": "record*", "fields": [
+                {"name": "v1", "type": "string"} ,
+                {"name": "v2", "type": "int*"} ] } },
+   { "name": "r4", "type": "tr2" }]
+}]]
+
+local nullable_exported = [[
+{"type":"record","fields":
+    [{"name":"r1","type":
+        {"type":"record","fields":
+            [{"name":"v1","type":"int"},
+            {"name":"v2","type":{"type":"string*"}}],
+        "name":"tr1"}},
+    {"name":"r2","type":"tr1*"},
+    {"name":"dummy","type":{"type":"array","items":"int"}},
+    {"name":"r3","type":
+        {"type":"record*","name":"tr2","fields":
+            [{"name":"v1","type":"string"},
+            {"name":"v2","type":{"type":"int*"}}]}},
+    {"name":"r4","type":"tr2"}],
+"name":"outer"}
+]]
+res = {schema.create(json.decode(nullable_orig))}
+test:is(res[1], true, "Schema created successfully")
+res = schema.export(res[2])
+test:is_deeply(res, json.decode(nullable_exported), "Exported schema is valid.")
```

Comment:
Are the original schema and expected to be exported the same in the case? Maybe encode it once here?